setup/nextcloud.sh

```diff
@@ -142,10 +152,12 @@ if [ ! -f $STORAGE_ROOT/owncloud/owncloud.db ]; then
   'overwritewebroot' => '/cloud',
   'overwrite.cli.url' => '/cloud',
   'user_backends' => array(
-	array(
-	  'class'=>'OC_User_IMAP',
-	  'arguments'=>array('{127.0.0.1:993/imap/ssl/novalidate-cert}')
-	)
+    array(
+      'class' => 'OC_User_IMAP',
+        'arguments' => array(
+          '127.0.0.1', 143, '', ''
```

Comment:
This works better for me: `array('$PRIMARY_HOSTNAME', 993, 'ssl')` (with only 3 items in array)

Please note:
I have entered my hostname directly instead of `$PRIMARY_HOSTNAME`, I have not tested if it works that way

Reply:
> 
> 
> This works better for me: `array('$PRIMARY_HOSTNAME', 993, 'ssl')` (with only 3 items in array) if you have users with different domains, your config works only for users of your primary domain.
> 
> Please note:
> I have entered my hostname directly instead of `$PRIMARY_HOSTNAME`, I have not tested if it works that way

See @JoshData's post above as to why we don't want to use $PRIMARY_HOSTNAME.

I use an additional domain for my email that is not $PRIMARY_HOSTNAME and I am able to login with the PR with multiple domains hosted by $PRIMARY_HOSTNAME.


https://github.com/nextcloud/user_external/blob/2fd648db4c0cb9c70d1df2026cadfff222244774/lib/imap.php#L34-L39

`array('127.0.0.1', 143)` and `array('127.0.0.1', 143, '', '')` should be functionally equivalent, with the prior being explicit.  I've tested both and both work fine.